x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java

```diff
@@ -659,6 +679,16 @@ LogEntryBuilder withRequestBody(RestRequest request) {
             return this;
         }
 
+        LogEntryBuilder withRequestId(String requestId, ThreadContext threadContext) {
+            if (Strings.isNullOrEmpty(requestId)) {
```

Comment:
This is again a place where I think we should not generate if missing. Missing is an indication that we don't know how this request came to be which is a problem in itself.

Reply:
We don't generate if missing - we just pull it from the thread context if it was passed in as null.
We can remove that (by making all callers do the extraction explicitly) if we want.

Reply:
Ahhhh, yes, it makes sense now. I was mistaken in my belief that they're also generated here. All precautionary recommendations in the other comments assumed they're also generated here.

All good, sorry.